### ISSUE TITLE
Questions about BART pretraining hyperparameters.

### ISSUE THREAD

**Issue by gorka96** [prajdabre/yanmtt#6]
Hei!

I want to train a BART base model for my language, and I'm trying to use the same hyper-parameters as in the BART base for English, and I have some doubts about some of the arguments I need to choose in your toolkit.

--batch_size 
This one is measured in tokens in your toolkit. Do you know if the 8K batch reported in the original BART paper refers to sequences or batches?

And then I'm not sure what is exactly the difference between  --max_length  and --max_src/tgt_length, and how I should use them.

Thanks

**Comment by prajdabre** [comment]
--batch_size can have different meanings depending on the flag: --batch_size_indicates_lines

If you pass this flag (--batch_size_indicates_lines) then --batch_size indicates number of examples in batch. So something in the range of 16 to 32 should be ok depending on the GPU capacity. If not then something like 2048 or 4096 will work. In the original BART paper 8k means number of examples. They do gradient accumulation to approximate large batch sizes which I also support via --multistep_optimizer_steps.

--max_length is the maximum length of the example beyond which it is truncated.

Ignore -max_src/tgt_length as they are used for joint BART and NMT training which you wont use.

Hope this helps.

**Comment by gorka96** [comment]
Thank you!
